### ISSUE TITLE
Support for S3 Object Tagging

### ISSUE THREAD

**Issue by lachilles** [boto/boto#3689]
Please consider adding support for object tagging, a new service that was announced at re:Invent in December 2016. 
Documentation can be found [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/object-tagging.html).
For context, the full presentation can be found [here](https://www.youtube.com/watch?v=Yx1y6hwxuuk).

**Comment by lachilles** [comment]
Actually, I may just tackle this as my first contribution to the repo :)

**Comment by eric-weaver** [comment]
Word of caution, I hacked around quickly with getting boto to work with Object tagging by just adding the `x-amz-tagging` header properly formatted directly to the PUT request. In my testing it increased upload response times by 25% for reasons unknown to me. I ended up switching out these isolated uploads to use of boto3 and response times dropped by 50% from the average boto S3 PUT without Object tagging. Didn't look into why this was the case for me, but it was clear for me that using boto3 here was better for my use case.
